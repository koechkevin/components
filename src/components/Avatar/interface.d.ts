import { AvatarProps } from 'antd/lib/avatar';
import { CSSProperties, ReactNode } from 'react';

enum ChatStatus {
  online = 'online',
  offline = 'offline',
}

export interface Props extends AvatarProps {
  src?: string;
  dot?: boolean;
  status?: ChatStatus | string;
  size?: number | 'xsmall' | 'small' | 'medium' | 'large' | 'default';
  color?: string;
}
