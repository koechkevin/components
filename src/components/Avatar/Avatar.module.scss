@import '../../styles/global';

.avatar {
  &:global(.ant-avatar-square) {
    border-radius: 8px;
  }

  &:global(.ant-avatar img) {
    background-color: $color-basic-light-gray;
  }
}
