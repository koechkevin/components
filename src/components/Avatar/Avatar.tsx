import { Avatar as AntdAvatar } from 'antd';
import React, { FC } from 'react';

import styles from './Avatar.module.scss';

import { Props } from './interface.d';

export const Avatar: FC<Props> = (props) => {
  const { children, className, icon, src, shape, size, style, color } = props;
  const avatarSizes = {
    small: 32,
    medium: 40,
    large: 80,
    default: 'default',
  };
  const avatarSize = size && (avatarSizes[size] || size);
  const placeholderIcon = src ? icon : 'user';
  const classNames: string = [styles.avatar, className ? className : ''].join(' ');
  const avatarColor = color && !src ? { backgroundColor: color } : {};

  return (
    <AntdAvatar
      src={src}
      shape={shape}
      size={avatarSize}
      className={classNames}
      style={{ minWidth: avatarSize, ...avatarColor, ...style }}
      icon={!children ? placeholderIcon : ''}
    >
      {children}
    </AntdAvatar>
  );
};

Avatar.defaultProps = {
  size: 'default',
};

export default Avatar;
