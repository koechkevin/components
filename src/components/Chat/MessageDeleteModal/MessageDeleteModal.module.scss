@import '../../../styles/global';

.deleteModal {
  .content {
    margin: 32px 0;
    padding: 8px;
    flex-wrap: nowrap;
    border-radius: 8px;
    overflow: hidden;
    border: 1px solid $color-gray-regular;

    @include respond-to('xs-l-landscape') {
      margin: 16px 0;
    }

    .creator {
      display: inline-block;
      height: 24px;
      line-height: 24px;
      font-weight: bold;
      color: $color-base-dark;
    }

    .createdAt {
      margin-left: 6px;
      color: rgba($color: $color-base-dark, $alpha: 0.6);
    }

    .text {
      margin-top: 8px;
      overflow-y: auto;
      word-break: break-word;
      max-height: 350px;

      @include respond-to('xs') {
        max-height: 60px;
      }

      @include respond-to('sm') {
        max-height: 180px;
      }
    }

    .message {
      padding-left: 16px;
      flex: 1 auto;
      overflow: hidden;
    }
  }
}
