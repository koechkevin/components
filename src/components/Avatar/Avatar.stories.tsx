import { number, text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react';
import { Col, Row, Typography } from 'antd';
import React from 'react';

import { Center } from '../Center';
import { ListItem } from '../SideBar/CompanyModal';
import Avatar from './Avatar';

import logo from '../../icons/logo.svg';
import mockLogo from '../../icons/mocks/company.png';

const src = 'https://s3.amazonaws.com/uifaces/faces/twitter/bowbrick/128.jpg';
const size = number('size', 200);
const initials = text('initials', 'AA');
const { Text } = Typography;
const companyList: ListItem[] = [
  { companyId: '123', avatarUrl: logo, name: 'Airbus', count: 4, avatarColor: '#fb6d77' },
  { companyId: '1234', avatarUrl: mockLogo, name: 'Pepsi', count: 21, avatarColor: '#46a094' },
  { companyId: '12345', avatarUrl: '', name: 'Airbus', avatarColor: '#5d85ce' },
];

storiesOf('Avatar', module)
  .addDecorator((story) => <Center>{story()}</Center>)
  .add('default', () => <Avatar />, { notes: 'Avatar default styles' })
  .add(
    'with circle shape and image',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>24</Text> xsmall{' '}
          </Col>
          <Col span={8}>
            <Text strong>40</Text> small{' '}
          </Col>
          <Col span={8}>
            <Text strong>80</Text> medium{' '}
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar src={src} size="small" />
          </Col>
          <Col span={8}>
            <Avatar src={src} size="medium" />
          </Col>
          <Col span={8}>
            <Avatar src={src} size="large" />
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Circle shape and image',
    },
  )
  .add(
    'with square shape and image',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>24</Text> xsmall
          </Col>
          <Col span={8}>
            <Text strong>40</Text> small
          </Col>
          <Col span={8}>
            <Text strong>80</Text> medium
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar src={src} shape="square" size="small" />
          </Col>
          <Col span={8}>
            <Avatar src={src} shape="square" size="medium" />
          </Col>
          <Col span={8}>
            <Avatar src={src} shape="square" size="large" />
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with square shape and image',
    },
  )
  .add(
    'with square shape and initials',
    () => (
      <Row>
        <Row>
          <Col span={12}>
            <Text strong>32</Text> small
          </Col>
          <Col span={12}>
            <Text strong>40</Text> large
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={12}>
            <Avatar size="small" shape="square" color="#46a094" style={{ fontWeight: 'bold', fontSize: 10 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={12}>
            <Avatar size="medium" shape="square" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Square shape and initials',
    },
  )
  .add(
    'with circle shape and initials',
    () => (
      <Row>
        <Row gutter={36}>
          <Col span={8}>
            <Text strong>32</Text> small
          </Col>
          <Col span={8}>
            <Text strong>40</Text> medium
          </Col>
          <Col span={8}>
            <Text strong>80</Text> large
          </Col>
        </Row>
        <Row gutter={36}>
          <Col span={8}>
            <Avatar size="small" shape="circle" color="#46a094" style={{ fontWeight: 'bold', fontSize: 10 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={8}>
            <Avatar size="medium" shape="circle" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
          <Col span={8}>
            <Avatar size="large" shape="circle" color="#8260da" style={{ fontWeight: 'bold', fontSize: 20 }}>
              {initials}
            </Avatar>
          </Col>
        </Row>
      </Row>
    ),
    {
      notes: 'Avatar with Square shape and initials',
    },
  )
  .add('with custom size', () => <Avatar src={src} size={size} />, {
    notes: 'Avatar with custom size',
  })
  .add('with an icon', () => <Avatar icon="user" />, {
    notes: 'Avatar with an icon',
  })
  .add('company logo', () => (
    <Row gutter={32}>
      {companyList.map((company) => (
        <Col span={8} key={company.companyId}>
          <Avatar src={company.avatarUrl} size="medium" shape="square" color={company.avatarColor}>
            {company.name[0]}
          </Avatar>
        </Col>
      ))}
    </Row>
  ));
