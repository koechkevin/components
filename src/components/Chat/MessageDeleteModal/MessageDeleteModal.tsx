import { Col, Row } from 'antd';
import { ModalProps } from 'antd/lib/modal';
import React, { createRef, FC, useEffect } from 'react';

import { MessageItemProps } from '../../../typings/Message';
import { FileExtension, getNameInitials, timestampFormat } from '../../../utils';
import { ProductType } from '../../../utils';
import { Avatar } from '../../Avatar';
import { FileItem } from '../../FileUpload';
import { Modal } from '../../Modal';
import { Text } from '../../Typography';

import styles from './MessageDeleteModal.module.scss';

interface Props extends ModalProps {
  message: MessageItemProps;
}

const MessageDeleteModal: FC<Props> = (props) => {
  const { visible, message, ...otherProps } = props;
  const { author, text, createdAt, file } = message || {};
  const isCandidate: boolean = author && author.productId === ProductType.Company ? false : true;
  const textRef = createRef<HTMLDivElement>();

  useEffect(() => {
    setTimeout(() => {
      if (textRef && textRef.current) {
        textRef.current.innerHTML = text || '';
      }
    }, 100);
  }, [visible, textRef, text]);

  return (
    <Modal
      centered
      width={500}
      okText="Delete"
      visible={visible}
      title="Delete Message"
      className={styles.deleteModal}
      okButtonProps={{ type: 'danger' }}
      {...otherProps}
    >
      <Row>
        <Row type="flex" justify="center">
          <Text style={{ textAlign: 'center' }}>
            Are you sure you want to delete this message?
            <br /> This cannot be undone
          </Text>
        </Row>
        <Row className={styles.content} type="flex">
          <Col>
            {author && (
              <Avatar
                size={40}
                src={author.avatar}
                shape={isCandidate ? 'square' : 'circle'}
                style={{ background: author.avatarColor }}
              >
                {getNameInitials(author.name)}
              </Avatar>
            )}
          </Col>
          <Col className={styles.message}>
            {author && (
              <Row>
                <span className={styles.creator}>{author.name} </span>
                <span className={styles.createdAt}>{createdAt && timestampFormat(new Date(createdAt))} </span>
              </Row>
            )}
            <Row className={styles.text}>
              <div id="text" ref={textRef} />
              {file && (
                <FileItem
                  {...file}
                  name={file.filename || file.name}
                  extension={file.type as FileExtension}
                  style={{ margin: '8px 0' }}
                />
              )}
            </Row>
          </Col>
        </Row>
      </Row>
    </Modal>
  );
};

export default MessageDeleteModal;
