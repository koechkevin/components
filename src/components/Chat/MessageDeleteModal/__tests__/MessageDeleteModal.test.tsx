import { mount } from 'enzyme';
import React from 'react';

import { Avatar } from '../../../Avatar';
import MessageDeleteModal from '../MessageDeleteModal';

describe('Tests for message delete modal', () => {
  const props = {
    visible: true,
    message: {
      id: 'sampleMessageId',
      createdAt: '2019-10-08 06:22:09',
      author: {
        id: 'user-fake-id',
        name: 'Jane Doe',
        avatarColor: '#4224',
        chatStatus: 'online',
        productId: 'company-portal',
      },
      text: 'sample text',
    },
  };

  const wrapper = mount(<MessageDeleteModal {...props} />);

  it('should mount the delete modal component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain a warning message', () => {
    expect(wrapper.contains('Are you sure you want to delete this message?')).toEqual(true);
  });

  it('should render author name', () => {
    expect(wrapper.contains(props.message.author.name)).toEqual(true);
  });

  it('should contain a circular avatar for company portal', () => {
    const avatar = wrapper.find(Avatar).first();

    expect(avatar.props().shape).toEqual('circle');
  });

  it('should contain square avatar for candidate portal', () => {
    const candidateProps = {
      ...props,
      message: {
        ...props.message,
        author: {
          ...props.message.author,
          productId: 'candidate-portal',
        },
      },
    };

    const candidateWrapper = mount(<MessageDeleteModal {...candidateProps} />);
    expect(
      candidateWrapper
        .find(Avatar)
        .first()
        .props().shape,
    ).toEqual('square');
  });
});
