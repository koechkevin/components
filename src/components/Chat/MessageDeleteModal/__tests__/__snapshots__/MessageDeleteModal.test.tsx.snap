// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests for message delete modal should mount the delete modal component 1`] = `ReactWrapper {}`;
